thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detector/metric/transfer/MetricTransfer.java

```diff
@@ -0,0 +1,34 @@
+package com.linkedin.thirdeye.detector.metric.transfer;
+
+import com.linkedin.thirdeye.api.MetricTimeSeries;
+import java.util.List;
+
+
+public class MetricTransfer {
+
+  /**
+   * Use the scaling factor to normalize the input time series
+   *
+   * @param metricsToModify the data MetricTimeSeries to be modified by scaling factor
+   * @param metricName the metricName withing the timeseries to be modified
+   * @param scalingFactorList list of scaling factors
+   * @return
+   */
+  public static void rescaleMetric(MetricTimeSeries metricsToModify, String metricName,
+      List<ScalingFactor> scalingFactorList) {
+    if (scalingFactorList == null || scalingFactorList.size() < 1) {
+      return;  // no transformation if there is no scaling factor in
+    }
+    // Went over the List of Scaling Factors and Rescale the timestamps with in the time range
+    // get the metric name
+    for (long ts : metricsToModify.getTimeWindowSet()) {
+      for (ScalingFactor sf: scalingFactorList) {
+        if (sf.getTimeWindow().contains(ts)) {
+          //update the metric value
+          metricsToModify.set(ts, metricName, metricsToModify.get(ts, metricName).doubleValue() * sf.getScalingFactor());
```

Comment:
You might want to set an upper or a lower bound for the scaled value. Since the information regarding history anomalies is limited in this function, we can just use the average of weeks for now.